packages/components/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,225 @@
+   */
+  maxWidth?: number | CSS.Property.MaxWidth;
+  /**
+   * It sets a delay period for the Tooltip
+   */
+  hideDelay?: number;
+  /**
+   * Function that will be called when target gets blurred
+   */
+  onBlur?: (evt: FocusEvent) => void;
+  /**
+   * Function that will be called when target gets focused
+   */
+  onFocus?: (evt: FocusEvent) => void;
+  /**
+   * Function that will be called when the user move the mouse out of the target
+   */
+  onMouseLeave?: (evt: MouseEvent) => void;
+  /**
+   * Function that will be called when the user move the mouse over of the target
+   */
+  onMouseOver?: (evt: MouseEvent) => void;
+  /**
+   * It sets the "preferred" position of the Tooltip
+   */
+  place?: TooltipPlace;
+  /**
+   * Class names to be appended to the className prop of the Tooltip’s target
+   */
+  targetWrapperClassName?: string;
+  /**
+   * An ID used for testing purposes applied as a data attribute (data-test-id)
+   */
+  testId?: string;
+  /**
+   * Boolean to control whether or not to render the tooltip in a React Portal.
+   * Rendering content inside a Portal allows the tooltip to escape the bounds
+   * of its parent while still being positioned correctly. Using a Portal is
+   * necessary if an ancestor of the tooltip hides overflow.
+   *
+   * Defaults to `false`
+   */
+  usePortal?: boolean;
+}
+
+export const Tooltip = ({
+  children,
+  className,
+  as: HtmlTag = 'span',
+  content,
+  id,
+  isVisible = false,
+  hideDelay = 0,
+  onBlur,
+  onFocus,
+  onMouseLeave,
+  onMouseOver,
+  targetWrapperClassName,
+  maxWidth = 360,
+  testId = 'cf-ui-tooltip',
+  place = 'auto',
+  usePortal = false,
+  ...otherProps
+}: TooltipProps) => {
+  const [show, setShow] = useState(false);
+
+  const elementRef = useRef(null);
+  const popperRef = useRef(null);
+  const [arrowRef, setArrowRef] = useState<HTMLSpanElement | null>(null);
+  const { styles: popperStyles, attributes, forceUpdate } = usePopper(
+    elementRef.current,
+    popperRef.current,
+    {
+      placement: place,
+      modifiers: [
+        {
+          name: 'arrow',
+          options: {
+            element: arrowRef,
+            padding: parseFloat(tokens.borderRadiusSmall),
+          },
+        },
+        {
+          name: 'offset',
+          options: {
+            offset: [0, 10],
+          },
+        },
+      ],
+    },
+  );
+
+  // necessary to update tooltip position in case the content is being updated
+  useEffect(() => {
+    if (forceUpdate !== null) {
+      forceUpdate();
+    }
+  }, [content, forceUpdate]);
+
+  const [isHoveringTarget, setIsHoveringTarget] = useState(false);
+  const [isHoveringContent, setIsHoveringContent] = useState(false);
+  useEffect(() => {
+    setShow(isHoveringContent || isHoveringTarget);
+  }, [isHoveringTarget, isHoveringContent]);
+
+  useEffect(() => {
+    if (isVisible) setShow(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why do we need this?

Reply:
this is something from the old component, I guess we can improve this part